api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +547,365 @@ paths:
           schema:
             $ref: '#/definitions/Problem'
 
+  /subscriptions:
+    post:
+      tags:
+        - high-level-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        This endpoint creates a subscription for EventTypes. The subscription is needed to be able to
+        consume events from EventTypes in a high level way when Nakadi stores the offsets and manages the
+        rebalancing of consuming clients.
+        The subscription is identified by its key parameters (owning_application, event_types, consumer_group). If
+        this endpoint is invoked several times with the same key subscription properties in body (order of even_types is
+        not important) - the subscription will be created only once and for all other calls it will just return
+        the subscription that was already created.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
+          schema:
+            $ref: '#/definitions/Subscription'
+          required: true
+      responses:
+        '200':
+          description: |
+            Subscription for such parameters already exists. Returns subscription object that already
+            existed.
+          schema:
+            $ref: '#/definitions/Subscription'
+          headers:
+            Content-Location:
+              description: the URI of existing subscription resource
+              type: string
+            Location:
+              description: the URI of existing subscription resource
+              type: string
+        '201':
+          description: Subscription was successfuly created. Returns subscription object that was created.
+          schema:
+            $ref: '#/definitions/Subscription'
+          headers:
+            Content-Location:
+              description: the URI of created subscription resource
+              type: string
+            Location:
+              description: the URI of created subscription resource
+              type: string
+        '400':
+          description: Bad Request
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+    get:
+      tags:
+        - high-level-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Lists all subscriptions that exist in a system. List is ordered by creation date/time descending (newest
+        subscriptions come first).
+      parameters:
+        - name: owning_application
+          in: query
+          description: |
+            Parameter to filter subscriptions list by owning application. If not specified - the result list will
+            contain subscriptions of all owning applications.
```

Comment:
We are not doing it now since the value of owning_application is user controlled and not automatically generated. So a default filtering would not be possible or could lead to empty replies in case the user provides a value for owning application that is different than the one that the IAM answers.